src/3rd_party/CMakeLists.txt

```diff
@@ -78,18 +84,29 @@ if(USE_SENTENCEPIECE)
 
   add_subdirectory(./sentencepiece)
   include_directories(./sentencepiece)
-
+  if(NOT COMPILE_DECODER_ONLY)
   set_target_properties(spm_encode spm_decode spm_train spm_normalize spm_export_vocab
     PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
+  endif()
 
   if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0))
+  if(COMPILE_DECODER_ONLY)
+    set_property(TARGET sentencepiece-static APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-class-memaccess")
```

Comment:
Why does disabling warnings apply only if we're compiling only the decoder?

Reply:
Actually, I am removing this.